src/platform/windows/mod.rs

```diff
@@ -0,0 +1,1482 @@
+use std::default::Default;
+use std::ffi::CString;
+use std::io::{Error, ErrorKind};
+use std::marker::{Send, Sync};
+use std::mem;
+use std::ops::Deref;
+use std::env;
+use std::ptr;
+use std::slice;
+
+use libc::{intptr_t, memcmp};
+
+use uuid::Uuid;
+
+use serde;
+use bincode;
+use bincode::serde::DeserializeError;
+
+use winapi;
+use winapi::{HANDLE, INVALID_HANDLE_VALUE, LPVOID};
+use kernel32;
+
+// some debug bump macros to better track what's going on in case of errors
+lazy_static! {
+    static ref CURRENT_PROCESS_ID: winapi::ULONG = unsafe { kernel32::GetCurrentProcessId() };
+    static ref CURRENT_PROCESS_HANDLE: intptr_t = unsafe { kernel32::GetCurrentProcess() as intptr_t };
+
+    static ref DD_ENABLED: bool = env::var_os("IPC_CHANNEL_WIN_DEBUG_DUMP").is_some();
+    static ref DD2_ENABLED: bool = env::var_os("IPC_CHANNEL_WIN_DEBUG_MORE_DUMP").is_some();
+}
+
+macro_rules! dd { ($($rest:tt)*) => { if *DD_ENABLED { println!($($rest)*); } } }
+macro_rules! dd2 { ($($rest:tt)*) => { if *DD_ENABLED || *DD2_ENABLED { println!($($rest)*); } } }
+
+// We stick these in WinErrors to indicate various things
+// that aren't quite errors.
+const MAGIC_CODE_BASE: u32 = 0xdeadbee0u32;
+const MAGIC_NO_OUTSTANDING_DATA_CODE: u32 = MAGIC_CODE_BASE + 0;
+const MAGIC_CHANNEL_CLOSED_CODE: u32 = MAGIC_CODE_BASE + 1;
+
+// When we create the pipe, how big of a write buffer do we specify?
+// This is reserved in the nonpaged pool.  The fragment size is the
+// max we can write to the pipe without fragmentation, and the
+// buffer size is what we tell the pipe it is, so we have room
+// for out of band data etc.  We can probably do way less than
+// 16k extra.
+const MAX_FRAGMENT_SIZE: usize = 64 * 1024;
+const WRITE_BUFFER_SIZE: usize = MAX_FRAGMENT_SIZE + 16*1024;
+
+// When we read, how big of a heap buffer do we allocate?
+const READ_BUFFER_SIZE: usize = 128 * 1024; //128k;
+const READ_BUFFER_MAX_GROWTH: usize = 1 * 1024 * 1024; // 1MB max growth
+
+// The message header is 2xu32 -- first one is the length of the data,
+// second one is the length of any out of band data.  If the first
+// field is INVALID_HEADER_DATA_SIZE, then the second one signifies
+// a control code (currently only MAGIC_CHANNEL_CLOSED_CODE is used)
+const HEADER_SIZE: usize = 8; //mem::size_of::<u32>()*2;
+const INVALID_HEADER_DATA_SIZE: u32 = 0xffffffffu32;
+
+#[allow(non_snake_case)]
+fn GetLastError() -> u32 {
+    unsafe {
+        kernel32::GetLastError()
+    }
+}
+
+pub fn channel() -> Result<(OsIpcSender, OsIpcReceiver),WinError> {
+    let mut receiver = try!(OsIpcReceiver::new());
+    let sender = try!(receiver.sender());
+    Ok((sender, receiver))
+}
+
+// If we have any channel handles or shmem segments,
+// then we'll send an OsIpcOutOfBandMessage after
+// the data message.
+#[derive(Debug)]
+struct OsIpcOutOfBandMessage {
+    channel_is_sender: Vec<bool>,
+    channel_handles: Vec<intptr_t>,
+    shmem_sizes: Vec<u64>,
+    shmem_handles: Vec<intptr_t>,
```

Comment:
At one point I was trying to avoid using serde, but I gave up on that right quick. So yeah this can be reworked (along with the handle bit stuff above).